### DISCUSSION TITLE
Deprecate Constructor Injection

### DISCUSSION THREAD

**Discussion by TonyValenti** [abpframework/abp#15968] [General] [answered]
Now that ABP supports AutoFac 7.0, constructor injection should be replaced with required property injection.  This will eliminate a bunch of useless boilerplate as it relates to constructors.

**Comment by maliming** [comment, marked as answer]
Autofac is an implementation that replaces default DI, we can't assume that users will use it.

**Comment by real-zony** [comment]
I'm afraid I don't think this would be a good option, as Required Properties are a specific concept in Autofac. If I were to replace the dependency injection component (DryIoc), many of ABP's core functionalities would no longer work properly.
